docs/purposes_yaml_files.md

```diff
@@ -0,0 +1,517 @@
+  id: 'hamilton_cherrypick_to_sample_dilution'
+- name: Download Concentration (ng/ul) CSV
+  id: concentrations_ngul
+```
+
+@note Not all CSV generation has been migrated under the exports controller. See :csv_template
+
+Default (Plate): [{ name: 'Download Concentration (nM) CSV', id: 'concentrations_nm' }]
+
+Default (Tube): []
+
+#### :csv_template
+
+String, either `'show_extended'`, `'show'` or leave undefined.
+
+Toggles which template gets used when someone requests PlatesController#show
+with an accept 'text/csv'. If left undefined, or set to nil, no link will be
+generated.
+
+```yaml
+:csv_template: 'show_extended'
+```
+
+@deprecated This is pretty much exclusively used for generating the show_extended
+            template in the ISC pipeline. :file_links is more flexible.
+
+Default: nil.
+
+#### :alternative_workline_identifier
+
+This attribute defines this plate purpose
+as an alternative labware that could be referred as a workline identifier
+while printing the barcode for our current plate barcode. This could be apply
+to distinguish between different workflows for plates when all of them have in
+common the same stock plate (RT ticket #683047)
+
+```yaml
+:alternative_workline_identifier: LB Lib PCR-XP
+```
+
+#### :label_template
+
+String, used to select an alternative {Labels::Base label} template, such as for
+printing QC labels. A list of valid label templates can be found in
+{file:config/label_templates.yml}
+
+If unspecified, falls back on the default label template for the given printer
+specified in default_pmb_templates in {file:config/label_templates.yml}.
+
+```yaml
+:label_template: plate_xp
+```
+
+#### :submission
+
+Hash, specifying:
+
+  template_name: The name of the submission template to use.
+
+  request_options: Valid request options hash to pass in to the submission, the
+                  exact keys required will depend on the submission template.
+
+Used by {WorkCompletion} to automatically build a downstream submission when
+the labware is charged and passed.
+
+```yaml
+:submission:
+  template_name: 'MiSeq for GBS'
+  request_options:
+    :read_length: 150
+```
+
+#### :merger_plate
+
+Boolean, set to true on plate where multiple smaller plates get merged together.
+Assists with extracting correct stock plate information when the plates get
+split apart further down the pipeline.
+
+Usually used in concert with a creator like {LabwareCreators::QuadrantStamp}
+
+```yaml
+:merger_plate: true
+```
+
+#### :warnings
+
+Hash, used to generate warnings to the user when expected conditions aren't met.
+Valid keys are `pcr_cycles_not_in` where the value should be an array of
+acceptable values.
+
+```yaml
+warnings:
+  pcr_cycles_not_in:
+    - 6
+```
+
+### Presenter/Creator specific configuration
+
+These options are only used for specific creators or presenters.
+
+#### :dilutions
+
+Used to define binning both for sorting samples by concentration, and for
+annotating the resulting wells.
```

Comment:
I think I'm not getting the meaning for this, what does it do with the samples by concentration? Binning like discarding volume?
By annotating, it means,  setting the value of new volume/concentration?

Reply:
Binning, as in grouping together.  It's a little vague, because the different classes use it differently, hence the links to the individual creators themselves.

I think by annotating here I meant within the presenter (The wells get the bins/concentration information shown)